Merge/foundation release/1.10.3

ziogaschr opened the pull request.

holiman committed:
consensus/ethash: less lookups of block data

fjl committed:
params: begin v1.10.3 release cycle

karalabe committed:
eth, les: drop support for eth/64, fix eth/66 tests

karalabe committed:
Merge pull request #22636 from karalabe/drop-eth64

eth, les: drop support for eth/64

balajipachai committed:
accounts: documentation fixes (#22645)

* replaces `an chance` with `a chance`
* replaces `SignHashWithPassphrase` with `SignTextWithPassphrase` as there was no SignHashWithPasspharse function in the file

holiman committed:
cmd/geth: add db-command to inspect freezer index (#22633)

This PR makes it easier to inspect the freezer index, which could be useful to investigate things like #22111

maxsam4 committed:
cmd/faucet: support testnet flags in the faucet (#22545)


Co-authored-by: Felix Lange <fjl@twurst.com>
Co-authored-by: Martin Holst Swende <martin@swende.se>

r1cs committed:
eth/fetcher: avoid spurious timer events at startup (#22652)


Co-authored-by: Felix Lange <fjl@twurst.com>

rjl493456442 committed:
core, eth: faster snapshot generation (#22504)

* eth/protocols: persist received state segments

* core: initial implementation

* core/state/snapshot: add tests

* core, eth: updates

* eth/protocols/snapshot: count flat state size

* core/state: add metrics

* core/state/snapshot: skip unnecessary deletion

* core/state/snapshot: rename

* core/state/snapshot: use the global batch

* core/state/snapshot: add logs and fix wiping

* core/state/snapshot: fix

* core/state/snapshot: save generation progress even if the batch is empty

* core/state/snapshot: fixes

* core/state/snapshot: fix initial account range length

* core/state/snapshot: fix initial account range

* eth/protocols/snap: store flat states during the healing

* eth/protocols/snap: print logs

* core/state/snapshot: refactor (#4)

* core/state/snapshot: refactor

* core/state/snapshot: tiny fix and polish

Co-authored-by: rjl493456442 <garyrong0905@gmail.com>

* core, eth: fixes

* core, eth: fix healing writer

* core, trie, eth: fix paths

* eth/protocols/snap: fix encoding

* eth, core: add debug log

* core/state/generate: release iterator asap (#5)

core/state/snapshot: less copy

core/state/snapshot: revert split loop

core/state/snapshot: handle storage becoming empty, improve test robustness

core/state: test modified codehash

core/state/snapshot: polish

* core/state/snapshot: optimize stats counter

* core, eth: add metric

* core/state/snapshot: update comments

* core/state/snapshot: improve tests

* core/state/snapshot: replace secure trie with standard trie

* core/state/snapshot: wrap return as the struct

* core/state/snapshot: skip wiping correct states

* core/state/snapshot: updates

* core/state/snapshot: fixes

* core/state/snapshot: fix panic due to reference flaw in closure

* core/state/snapshot: fix errors in state generation logic + fix log output

* core/state/snapshot: remove an error case

* core/state/snapshot: fix condition-check for exhausted snap state

* core/state/snapshot: use stackTrie for small tries

* core/state/snapshot: don't resolve small storage tries in vain

* core/state/snapshot: properly clean up storage of deleted accounts

* core/state/snapshot: avoid RLP-encoding in some cases + minor nitpicks

* core/state/snapshot: fix error (+testcase)

* core/state/snapshot: clean up tests a bit

* core/state/snapshot: work in progress on better tests

* core/state/snapshot: polish code

* core/state/snapshot: fix trie iteration abortion trigger

* core/state/snapshot: fixes flaws

* core/state/snapshot: remove panic

* core/state/snapshot: fix abort

* core/state/snapshot: more tests (plus failing testcase)

* core/state/snapshot: more testcases + fix for failing test

* core/state/snapshot: testcase for malformed data

* core/state/snapshot: some test nitpicks

* core/state/snapshot: improvements to logging

* core/state/snapshot: testcase to demo error in abortion

* core/state/snapshot: fix abortion

* cmd/geth: make verify-state report the root

* trie: fix failing test

* core/state/snapshot: add timer metrics

* core/state/snapshot: fix metrics

* core/state/snapshot: udpate tests

* eth/protocols/snap: write snapshot account even if code or state is needed

* core/state/snapshot: fix diskmore check

* core/state/snapshot: review fixes

* core/state/snapshot: improve error message

* cmd/geth: rename 'error' to 'err' in logs

* core/state/snapshot: fix some review concerns

* core/state/snapshot, eth/protocols/snap: clear snapshot marker when starting/resuming snap sync

* core: add error log

* core/state/snapshot: use proper timers for metrics collection

* core/state/snapshot: address some review concerns

* eth/protocols/snap: improved log message

* eth/protocols/snap: fix heal logs to condense infos

* core/state/snapshot: wait for generator termination before restarting

* core/state/snapshot: revert timers to counters to track total time

Co-authored-by: Martin Holst Swende <martin@swende.se>
Co-authored-by: Péter Szilágyi <peterke@gmail.com>

karalabe committed:
core/types: drop some relice data types

karalabe committed:
Merge pull request #22666 from karalabe/remove-stale-datatype

core/types: drop some relice data types

karalabe committed:
all: make logs a bit easier on the eye to digest (#22665)

* all: add thousandths separators for big numbers on log messages

* p2p/sentry: drop accidental file

* common, log: add fast number formatter

* common, eth/protocols/snap: simplifty fancy num types

* log: handle nil big ints

karalabe committed:
eth/protocols/snap: use ephemeral channels to avoid cross-sync delveries

meowsbits committed:
core: add TestGenesisHashes and fix YoloV3 (#22559)

This adds simple unit test checking if the hard-coded genesis
hash values in package params match the actual genesis block hashes.

fjl committed:
log: fix formatting of big.Int (#22679)

* log: fix formatting of big.Int

The implementation of formatLogfmtBigInt had two issues: it crashed when
the number was actually large enough to hit the big integer case, and
modified the big.Int while formatting it.

* log: don't call FormatLogfmtInt64 for int16

* log: separate from decimals back, not front

Co-authored-by: Péter Szilágyi <peterke@gmail.com>

karalabe committed:
Merge pull request #22678 from karalabe/snap-ephemeral-channels

eth/protocols/snap: use ephemeral channels to avoid cross-sync delveries

zsfelfoldi committed:
les/vflux/server: fix priority cornercase causing fuzzer timeout (#22650)

* les/vflux/server: fix estimatePriority corner case

* les/vflux/server: simplify inactiveAllowance == 0 case

holiman committed:
trie: make stacktrie not mutate input values (#22673)

The stacktrie is a bit un-untuitive, API-wise: since it mutates input values.
Such behaviour is dangerous, and easy to get wrong if the calling code 'forgets' this quirk. The behaviour is fixed by this PR, so that the input values are not modified by the stacktrie. 

Note: just as with the Trie, the stacktrie still references the live input objects, so it's still _not_ safe to mutate the values form the callsite.

holiman committed:
core/state/snapshot: avoid copybytes for stacktrie

gballet committed:
eth/catalyst: add catalyst API prototype (#22641)

This change adds the --catalyst flag, enabling an RPC API for eth2 integration.
In this initial version, catalyst mode also disables all peer-to-peer networking.

Co-authored-by: Mikhail Kalinin <noblesse.knight@gmail.com>
Co-authored-by: Felix Lange <fjl@twurst.com>

karalabe committed:
Merge pull request #22686 from holiman/minor_fixes

core/state/snapshot: avoid copybytes for stacktrie

fjl committed:
cmd/devp2p: add support for -limit option in nodeset filter command (#22694)

The new -limit option makes the filter operate on top N nodes by score.
This also adds ENR attribute stats in the nodeset info command.
Node set commands are now documented in README.

fjl committed:
cmd/devp2p: add dns nuke-route53 command (#22695)

rjl493456442 committed:
core: nuke legacy snapshot supporting (#22663)

gballet committed:
ethash: no block reward in catalyst mode (#22697)

holiman committed:
trie: make stacktrie support binary marshal/unmarshal (#22685)

gballet committed:
go.mod: upgrade gopsutils to v3.21.4 (#22693)

This fixes the OpenBSD/arm64 build.

holiman committed:
tests: disable blockchain tests based on general state tests (#22704)

rjl493456442 committed:
eth, internal: extend the TraceCall API (#22245)

Adds an an optional parameter `overrides *map[common.Address]account` to the `eth_call` API in order for the caller to  can customize the state.

rjl493456442 committed:
eth/tracers, internal/ethapi: fix typos causing lint issue (#22711)

holiman committed:
les: fix goroutine leaks in tests (#22707)

holiman committed:
trie: improve the node iterator seek operation (#22470)

This change improves the efficiency of the nodeIterator seek
operation. Previously, seek essentially ran the iterator forward
until it found the matching node. With this change, it skips
over fullnode children and avoids resolving them from the database.

holiman committed:
accounts/external, signer/core: add support for EIP-2930 transactions (#22585)

This adds support for signing EIP-2930 with clef.

Commit:
rpc: add HTTPError type for HTTP error responses (#22677)

The new error type is returned by client operations contains details of
the response error code and response body.

Co-authored-by: Felix Lange <fjl@twurst.com>

karalabe committed:
eth/protocols, prp/tracker: add support for req/rep rtt tracking (#22608)

* eth/protocols, prp/tracker: add support for req/rep rtt tracking

* p2p/tracker: sanity cap the number of pending requests

* pap/tracker: linter <3

* p2p/tracker: disable entire tracker if no metrics are enabled

renaynay committed:
cmd/devp2p/internal/ethtest: run test suite as Go unit test (#22698)

This change adds a Go unit test that runs the protocol test suite
against the go-ethereum implementation of the eth protocol.

holiman committed:
core/state/snapshot, true: reuse dirty data instead of hitting disk when generating (#22667)

* core/state/snapshot: reuse memory data instead of hitting disk when generating

* trie: minor nitpicks wrt the resolver optimization

* core/state/snapshot, trie: use key/value store for resolver

* trie: fix linter

Co-authored-by: Péter Szilágyi <peterke@gmail.com>

renaynay committed:
cmd/devp2p/internal/ethtest: add more tx propagation tests (#22630)

This adds a test for large tx announcement messages, as well as a test to
check that announced tx hashes are requested by the node.

nisdas committed:
p2p/discover: improve discv5 handling of IPv4-in-IPv6 addresses (#22703)

When receiving PING from an IPv4 address over IPv6, the implementation sent
back a IPv4-in-IPv6 address. This change makes it reflect the IPv4 address.

holiman committed:
core: remove old conversion to shuffle leveldb blocks into ancients

holiman committed:
core/rawdb: fix datarace in freezer (#22728)

The Append / truncate operations were racy. When a datafile reaches 2Gb, a new file is needed. For this operation, we require a writelock, which is not needed in the 99.99% of all cases where the data does fit in the current head-file.

This transition from readlock to writelock was incorrect, and as the readlock was released, a truncate operation could slip in between, and truncate the data. This would have been fine, however, the Append operation continued writing as if no truncation had occurred, e.g writing item 5 where item 0 should reside.

This PR changes the behaviour, so that if when we run into the situation that a new file is needed, it aborts, and retries, this time with a writelock.

The outcome of the situation described above, running on this PR, would instead be that the Append operation exits with a failure.

rjl493456442 committed:
les: polish code (#22625)

* les: polish code

* les/vflus/server: fixes

* les: fix lint

fjl committed:
build: upgrade to golangci-lint v1.39.0 (#22696)

* build: upgrade to golangci-lint v1.39.0

* consensus/ethash: fix go vet warning regarding reflect.SliceHeader

* eth/catalyst: fix lint issue

* consensus/ethash: fix bug in memoryMapFile

karalabe committed:
Merge pull request #21467 from holiman/minor_ethashfix

consensus/ethash: less lookups of block data

lightclient committed:
cmd/puppeth: add support for authentication via ssh agent (#22634)

fjl committed:
build: upgrade -dlgo version to Go 1.16.3 (#22746)

holiman committed:
core/vm: make gas cost reporting to tracers correct (#22702)

Previously, the makeCallVariantGasCallEIP2929 charged the cold account access cost directly, leading to an incorrect gas cost passed to the tracer from the main execution loop.
This change still temporarily charges the cost (to allow for an accurate calculation of the available gas for the call), but then afterwards refunds it and instead returns the correct total gas cost to be then properly charged in the main loop.

karalabe committed:
eth/protocols/snap: generate storage trie from full dirty snap data (#22668)

* eth/protocols/snap: generate storage trie from full dirty snap data

* eth/protocols/snap: get rid of some more dead code

* eth/protocols/snap: less frequent logs, also log during trie generation

* eth/protocols/snap: implement dirty account range stack-hashing

* eth/protocols/snap: don't loop on account trie generation

* eth/protocols/snap: fix account format in trie

* core, eth, ethdb: glue snap packets together, but not chunks

* eth/protocols/snap: print completion log for snap phase

* eth/protocols/snap: extended tests

* eth/protocols/snap: make testcase pass

* eth/protocols/snap: fix account stacktrie commit without defer

* ethdb: fix key counts on reset

* eth/protocols: fix typos

* eth/protocols/snap: make better use of delivered data (#44)

* eth/protocols/snap: make better use of delivered data

* squashme

* eth/protocols/snap: reduce chunking

* squashme

* eth/protocols/snap: reduce chunking further

* eth/protocols/snap: break out hash range calculations

* eth/protocols/snap: use sort.Search instead of looping

* eth/protocols/snap: prevent crash on storage response with no keys

* eth/protocols/snap: nitpicks all around

* eth/protocols/snap: clear heal need on 1-chunk storage completion

* eth/protocols/snap: fix range chunker, add tests

Co-authored-by: Péter Szilágyi <peterke@gmail.com>

* trie: fix test API error

* eth/protocols/snap: fix some further liter issues

* eth/protocols/snap: fix accidental batch reuse

Co-authored-by: Martin Holst Swende <martin@swende.se>

holiman committed:
p2p/tracker: properly clean up fulfilled requests

karalabe committed:
Merge pull request #22751 from holiman/tracker_fix

p2p/tracker: properly clean up fulfilled requests

karalabe committed:
p2p/tracker: only reschedule wake if previous didn't run

karalabe committed:
Merge pull request #22753 from karalabe/p2p-tracker-stopfix

p2p/tracker: only reschedule wake if previous didn't run

holiman committed:
cmd/devp2p, eth/protocols/eth: fix tests + make sanity checks earlier (#22749)

GregTheGreek committed:
eth/gasprice: improve stability of estimated price (#22722)

This PR makes the gas price oracle ignore transactions priced at `<=1 wei`.

gbotrel committed:
tests/fuzzers: crypto/bn256 and crypto/bls12381 tests against gnark-crypto  (#22755)

Add more cross-fuzzers to fuzz bls with gnark versus geth's own bls12-381 library

rjl493456442 committed:
les, tests: fix les clientpool (#22756)

* les, tests: fix les clientpool

* tests: disable debug mode

* les: polish code

karalabe committed:
eth/protocols/snap: lower the packet size to avoid overloading link

renaynay committed:
cmd/devp2p: fix flaky SameRequestID test (#22754)

karalabe committed:
Merge pull request #22761 from karalabe/snap-small-packets

eth/protocols/snap: lower the packet size to avoid overloading link

holiman committed:
trie: remove redundant returns + use stacktrie where applicable (#22760)

* trie: add benchmark for proofless range

* trie: remove unused returns + use stacktrie

karalabe committed:
core, eth, ethdb, trie: simplify range proofs

karalabe committed:
Merge pull request #22762 from karalabe/snap-lower-complexity

core, eth, ethdb, trie: simplify range proofs

karalabe committed:
eth: restore eth_hashrate API endpoint

karalabe committed:
Merge pull request #22765 from karalabe/revert-eth-hashrate

eth: restore eth_hashrate API endpoint

karalabe committed:
core, eth: abort snapshot generation on snap sync and resume later

protolambda committed:
catalyst: check if block exists in assemble-block call with unknown parent-hash (#22770)

gballet committed:
add myself as code owner for catalyst (#22778)

aaronbuchwald committed:
docs: fix docstring on read head block (#22776)

chfast committed:
evm: remove unused errors left after EIP-2315 removal (#22767)

ligi committed:
github: add note about screenshots in issue template (#22764)

aaronbuchwald committed:
core/types: add license header (#22781)

aaronbuchwald committed:
core/vm: fix typo in comment (#22785)

aaronbuchwald committed:
core: remove unused else branch in reorg (#22783)

aaronbuchwald committed:
core/vm: replace repeated string with variable in tests (#22774)

aaronbuchwald committed:
core: fix typo in comment (#22773)

holiman committed:
README.md: update commands table, add note about web3.js version (#22748)

aaronbuchwald committed:
eth/filters: fix comment on PublicFilterAPI timeoutLoop (#22782)

aaronbuchwald committed:
core/state: remove toAddr helper in tests (#22772)

karalabe committed:
Merge pull request #22777 from karalabe/snapshots-abort-resume-on-sync

core, eth: abort snapshot generation on snap sync and resume later

karalabe committed:
eth/protocols/snap: use storage batch, not account batch in st task

karalabe committed:
Merge pull request #22789 from karalabe/snap-fix-batch

eth/protocols/snap: use storage batch, not account batch in st task

renaynay committed:
cmd/devp2p: fix flakey tests in CI (#22757)

This PR fixes a couple of issues in the eth test suite that caused flakiness when run in the CI.

chfast committed:
core/vm: clean up contract creation error handling (#22766)

Do not keep separate flag for "max code size exceeded" case, but instead
assign appropriate error for it sooner.

aaronbuchwald committed:
core/vm: fix interpreter comments (#22797)

* Fix interpreter comment

* Fix comment

fjl committed:
params: remove dependency on crypto (#22788)

* params: remove dependency on crypto

Package params should not depend on package crypto because building
crypto requires cgo.

Since build/ci.go needs package params to get the go-ethereum version
number, C code must be compiled in order to run the build tool, which is
annoying for certain cross-compilation setups.

* params: add SectionHead

MariusVanDerWijden committed:
cmd/utils: don't crash on nonexistent datadir (#22738)

karalabe committed:
Merge pull request #22739 from holiman/remove_code

core: remove old conversion to shuffle leveldb blocks into ancients

karalabe committed:
eth: don't print db upgrade warning on db init

karalabe committed:
Merge pull request #22803 from karalabe/silence-scary-warning

eth: don't print db upgrade warning on db init

fjl committed:
cmd/utils: use eth DNS tree for snap discovery (#22808)

This removes auto-configuration of the snap.*.ethdisco.net DNS discovery tree.
Since measurements have shown that > 75% of nodes in all.*.ethdisco.net support
snap, we have decided to retire the dedicated index for snap and just use the eth
tree instead.

The dial iterators of eth and snap now use the same DNS tree in the default configuration,
so both iterators should use the same DNS discovery client instance. This ensures that
the record cache and rate limit are shared. Records will not be requested multiple times.

While testing the change, I noticed that duplicate DNS requests do happen even
when the client instance is shared. This is because the two iterators request the tree
root, link tree root, and first levels of the tree in lockstep. To avoid this problem, the
change also adds a singleflight.Group instance in the client. When one iterator
attempts to resolve an entry which is already being resolved, the singleflight object
waits for the existing resolve call to finish and returns the entry to both places.

fjl committed:
build: improve cross compilation setup (#22804)

This PR cleans up the CI build system and fixes a couple of issues.

- The go tool launcher code has been moved to internal/build. With the new
  toolchain functions, the environment of the host Go (i.e. the one that built
  ci.go) and the target Go (i.e. the toolchain downloaded by -dlgo) are isolated
  more strictly. This is important to make cross compilation and -dlgo work
  correctly in more cases.
- The -dlgo option now skips the download and uses the host Go if the running Go
  version matches dlgoVersion exactly.
- The 'test' command now supports -dlgo, -cc and -arch. Running unit tests with
  foreign GOARCH is occasionally useful. For example, it can be used to run
  32-bit tests on Windows. It can also be used to run darwin/amd64 tests on
  darwin/arm64 using Rosetta 2.
- The 'aar', 'xcode' and 'xgo' commands now use a slightly different method to
  install external tools. They previously used `go get`, but this comes with the
  annoying side effect of modifying go.mod. They now use `go install` instead,
  which is the recommended way of installing tools without modifying the local
  module.
- The old build warning about outdated Go version has been removed because we're
  much better at keeping backwards compatibility now.

fjl committed:
go.mod: go mod tidy (#22814)

This updates go.mod for the addition of golang.org/x/sync.

fjl committed:
build: fix iOS framework build (#22813)

This fixes a regression introduced in #22804.

fjl committed:
appveyor.yml: upgrade to VisualStudio 2019 image (#22811)

fjl committed:
build: fix windows installer build for NSIS v3.05 (#22821)

With the update to a newer AppVeyor build image, creating the Windows
installer no longer worked because of a string quoting error in EnvVarUpdate.nsh.

This applies the fix recommended in https://stackoverflow.com/questions/62081765.

renaynay committed:
cmd/devp2p/internal/ethtest: send simultaneous requests on one connection (#22801)

This changes the SimultaneousRequests test to send the requests from the same
connection, as it doesn't really make sense to test whether a node can respond
to two requests with different request IDs from separate connections.

fjl committed:
params: go-ethereum v1.10.3 stable

ziogaschr committed:
Merge branch 'foundation/1.10.3' into merge/foundation-release/1.10.3

# Conflicts:
#	.github/CODEOWNERS
#	README.md
#	appveyor.yml
#	build/ci.go
#	cmd/devp2p/internal/ethtest/chain.go
#	cmd/faucet/faucet.go
#	cmd/geth/config.go
#	cmd/puppeth/ssh.go
#	cmd/utils/flags.go
#	consensus/ethash/consensus.go
#	core/blockchain.go
#	core/genesis_test.go
#	core/vm/evm.go
#	core/vm/instructions_test.go
#	core/vm/interpreter.go
#	eth/tracers/api.go
#	go.mod
#	go.sum
#	params/config.go
#	params/version.go

meowsbits committed:
go.mod,go.sum: fix: no required module provides package github.com/tidwall/gjson

Date: 2021-05-12 13:02:50-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/devp2p/internal/ethtest: type ChainConfig not declared by package params

Date: 2021-05-12 13:08:17-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/faucet: no new variables on left side of :=

Date: 2021-05-12 13:11:27-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/catalyst: TxGas not declared by package params/

Date: 2021-05-12 13:14:06-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/catalyst: core.Genesis -> genesisT.Genesis, other import/cg tweaks

Date: 2021-05-12 13:16:45-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth: missing return (typecheck)

Date: 2021-05-12 13:18:18-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
go.mod,go.sum: go mod tidy

Date: 2021-05-12 13:19:20-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/coregeth,params/types/ctypes,params/types/genesisT,params/types/goethereum,params/types/multigeth,params/types/parity: init Get/SetCatalystBlock for configurator iface

Date: 2021-05-12 13:34:15-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/catalyst: use GetCatalystTransition method for configurator

Date: 2021-05-12 13:35:22-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/parity: fixup! catalystBlock -> CatalystBlock

Date: 2021-05-12 13:36:02-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: mismatched types (int and uint64)

Date: 2021-05-12 13:37:43-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
consensus/ethash: config.IsCatalyst undefined

Date: 2021-05-12 13:41:50-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/tracers: fix import paths

Date: 2021-05-12 13:43:59-05:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
cleanup comment notes

ziogaschr commented on core/blockchain.go:
```diff
@@ -1231,74 +1216,19 @@ func (bc *BlockChain) InsertReceiptChain(blockChain types.Blocks, receiptChain [
-					deleted = deleted[0:]
+				return i, fmt.Errorf("containing header #%d [%x..] unknown", block.Number(), block.Hash().Bytes()[:4])
+			}
+			// TODO(ziogaschr): check removal of https://github.com/etclabscore/core-geth/commit/f71d78cb26aad5bc7e4df76fadd082b69ca2de82
```
@meowsbits can you please check if you want this logic maintained? I think with the latest upstream changes it is not needed.

meowsbits replied:
- `core: remove old conversion to shuffle leveldb blocks into ancients ` https://github.com/ethereum/go-ethereum/commit/83375b08731d95fdc91b6e1576db596c21384a3d
- https://github.com/ethereum/go-ethereum/pull/22739

LOKTM :heavy_check_mark:

ziogaschr replied:
Resolved by 848cd11

ziogaschr committed:
eth: drop support for eth/64 in DefaultProtocolVersions

ziogaschr committed:
eth/tracers: use TraceCallConfig on tracer package (OE compatibility layer)

meowsbits committed:
go.mod: go-openrpc-reflect -> 0.35 (unbump)

Date: 2021-05-13 08:40:43-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
go.mod,go.sum: go mod tidy

Date: 2021-05-13 08:40:59-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
go.mod: unbump openrpc/meta_schema

Date: 2021-05-13 08:41:42-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
go.sum: go mod tidy

Date: 2021-05-13 08:41:56-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/devp2p/internal/ethtest: no field or method ChainID

Date: 2021-05-13 08:43:44-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Revert "eth: missing return (typecheck)"

This reverts commit 68a6a97d98f8e1e66d7a46397a8533aeb0e7a7ff.

meowsbits committed:
cmd/devp2p/internal/ethtest: add debug data for unexpected negotiated protocol version

Date: 2021-05-13 09:30:47-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/devp2p/internal/ethtest: fix missing DefaultProtocolVersions assignment in test config

Date: 2021-05-13 09:43:03-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/utils: set default for --eth.protocols flag

Date: 2021-05-13 09:46:47-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params: fix yolov3 genesis hash hardcode

Date: 2021-05-13 12:22:20-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
ethclient: eth_hashrate is back

Date: 2021-05-13 12:28:51-05:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
core: remove comment

ziogaschr committed:
eth/tracers: disable traceCall test vector

ziogaschr committed:
cmd/utils: change help message for `—eth.protocols` flag

ziogaschr committed:
cmd/utils: remove not needed check for empty `—eth.protocols` flag value, as we set a default

meowsbits commented on eth/tracers/api_test.go:
```diff
@@ -362,19 +362,21 @@ func TestOverridenTraceCall(t *testing.T) {
+		// 		Tracer: &tracer,
+		// 	},
+		// 	expectErr: core.ErrInsufficientFundsForTransfer,
+		// 	expect:    nil,
```
Would it make sense to modify the test to `expectErr: nil; expect: <blah>`? (but leaving a comment about the difference, as you already have done)

ziogaschr replied:
Thought of that too, but this test, wanna test the override functionality. Testing what you described, is being covered already in the tracers tests.

meowsbits committed:
Merge pull request #379 from etclabscore/merge/foundation-release/1.10.3-ethprotocols-flag-default

cmd/utils: set default for --eth.protocols flag

ziogaschr committed:
cmd/utils: use DefaultProtocolVersions from flag

# Conflicts:
#	cmd/utils/flags.go

meowsbits commented on cmd/utils/flags.go:
```diff
@@ -1738,37 +1738,36 @@ func SetEthConfig(ctx *cli.Context, stack *node.Node, cfg *ethconfig.Config) {
-					break
-				}
+		isValid := false
+		for _, proto := range eth.DefaultProtocolVersions {
```
This validates the passed protocol version using `DefaultProtocolVersions` as the list of valid options (rather than as -- one might assume -- a list of _defaults_). Is this desired? Is there a list of eligibile-but-not-necessarily-default versions?

ziogaschr replied:
Nice catch. Indeed the `eth.DefaultProtocolVersions` is misleading here. 
The intention was, that this value holds all the supported networks, the same way the used to have in the protocols.go (go-ethereum, till recently), and is being used as the Default value for the flag. 
So we have to either rename it or use a second var for it. 
I am on this one...

ziogaschr replied:
Resolved by 15e0f16

ziogaschr committed:
cmd/devp2p/internal/ethtest, cmd/utils, eth: rename `eth.DefaultProtocolVersions` to `eth.SupportedProtocolVersions` for readability

ziogaschr committed:
make SupportedProtocolVersions chain-dependent

ziogaschr committed:
params/types: add `omitempty` to json ChainConfig property `supportedProtocolVersions`

meowsbits committed:
params/types/goethereum: example implementation for default protocol versions

Date: 2021-05-20 07:54:24-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/ethconfig,params/types/goethereum,params/vars: create vars.DefaultProtocolVersions

This creates a 'default'-namespaced value
that defines default values; this allows chain
configurators or other defaults to use explicitly
'default' values as defaults, rather than awkwardly
using vars.SupportedProtocolVersions as their
default values.

Date: 2021-05-20 08:02:43-05:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
params/types/multigeth: set default protocol versions

ziogaschr committed:
cmd/devp2p/internal/ethtest: use vars.DefaultProtocolVersions

meowsbits committed:
params: use DefaultProtocolVersions for chainconfig values

Date: 2021-05-21 05:47:32-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/coregeth,params/types/multigeth,params/types/parity: install ProtocolVersion defaults for configurator implementations

Date: 2021-05-21 05:56:30-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/utils: fixup duplicating protocol version configs

Protocol versions were being appending to the configuration
in duplicate, since the cfg value passed to the function
is the ethconfig.Defaults value.

This cleans the incoming cfg.ProtocolVersions field before
appending to it, and adds an Info log line.

Date: 2021-05-21 06:35:48-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/utils: modify fatal log text to note that value invalid

Date: 2021-05-21 06:38:13-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/coregeth,params/types/goethereum,params/types/multigeth,params/types/parity: persist cnfgrtr defaults to cfg data during get

ProtocolVersion getters now persist the defaults
to their chain configuration if the value was previously
empty.

This allows the configuration structs to be
compared at the schema level, and to handle default
values equivalently.

Date: 2021-05-24 06:30:54-05:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Merge pull request #384 from etclabscore/merge/foundation-release/1.10.3-protocolVersions-chain-dependent

Merge/foundation release/1.10.3 protocol versions chain dependent
